pkg/controller/broker/broker_controller.go

```diff
@@ -380,6 +380,23 @@ func (r *ReconcileBroker) getBrokerStatefulSet(broker *rocketmqv1alpha1.Broker,
 		statefulSetName = broker.Name + "-" + strconv.Itoa(brokerGroupIndex) + "-replica-" + strconv.Itoa(replicaIndex)
 	}
 
+	var affinity corev1.Affinity
+	if broker.Spec.NodeAffinityKey != "" && broker.Spec.NodeAffinityOperator != "" && broker.Spec.NodeAffinityValues != nil {
```

Comment:
Have you tried if the user do not config Affinity and what happens when affinity is nil ?